lib/urlwatch/storage.py

```diff
@@ -600,6 +602,19 @@ def get_history_data(self, guid, count=1):
                     break
         return history
 
+    def has_history_data(self, guid):
+        if not self._cached_has_history_data_set:
+            self._cached_has_history_data_set = frozenset(guid[0] for guid in
+                                                           list(CacheEntry.query(self.db,CacheEntry.c.guid,
+                                                                                 where=((CacheEntry.c.tries == 0) |
+                                                                                        (CacheEntry.c.tries == None))))) # noqa:E711
```

Comment:
Wouldn't we just need to `SELECT DISTINCT guid FROM CacheEntry`? (untested, but probably working...)

```suggestion
        if not self._cached_has_history_data_set:
            self._cached_has_history_data_set = frozenset(CacheEntry.query(self.db, CacheEntry.c.guid.distinct("guid")))
```

Reply:
Alternatively, drop the whole caching thing and just query the count directly:

```suggestion
        return next(CacheEntry.query(self.db, minidb.func.count("*")("count"), where=CacheEntry.c.guid == guid)).count > 0
```

Reply:
The first works but it still needs to loop over the result since its a "RowProxy" otherwise. 

`self._cached_has_history_data_set = frozenset(guid[0] for guid in list(CacheEntry.query(self.db, CacheEntry.c.guid.distinct("guid"))))  # noqa:E711` seems to work.

The second works but it is just as slow as using get_history_data() direct
`Benchmark: Queue 4059 Jobs took 0:02:07.422035` and cached 
`Benchmark: Queue 4059 Jobs took 0:00:00.194317`

Reply:
I did try some other cases with the first but it won't work. 
If a Job failed, i still has its guid in the database with no data and a try count greater than 0